modules/snmp/init.go

```diff
@@ -9,6 +9,19 @@ import (
 
 var snmpHandler = gosnmp.NewHandler
 
+func generateSNMPv3MsgFlags(level int) gosnmp.SnmpV3MsgFlags {
+	var flag gosnmp.SnmpV3MsgFlags
+	switch level {
+	case 1:
+		flag = gosnmp.NoAuthNoPriv
+	case 2:
+		flag = gosnmp.AuthNoPriv
+	case 3:
+		flag = gosnmp.AuthPriv
+	}
+	return gosnmp.SnmpV3MsgFlags(flag)
```

Comment:
```suggestion
	return flag
```

I think we don't need this cast.